### PR TITLE
eclint: init at 0.3.4

### PR DESCRIPTION
**Pull request by lucperkins**
###### Description of changes

Adds the [eclint](https://gitlab.com/greut/eclint) tool to Nixpkgs (specifically the version written in Go).

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/eclint/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, buildGoModule
+, fetchFromGitLab
+}:
+
+buildGoModule
+rec {
+  pname = "eclint";
+  version = "0.3.3";
```
Comment (zowoq) [inline pkgs/development/tools/eclint/default.nix:9]: https://gitlab.com/greut/eclint/-/releases/v0.3.4

File: pkgs/development/tools/eclint/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    owner = "greut";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-9i2oAqFXflWGeBumE/5njaafBRhuRQSbA/ggUS72fwk=";
+  };
+
+  vendorSha256 = "sha256-XAyHy7UAb2LgwhsxaJgj0Qy6ukw9szeRC9JkRb+zc0Y=";
+
+  meta = with lib; {
+    homepage = "https://github.com/greut/eclint";
```
Comment (zowoq) [inline pkgs/development/tools/eclint/default.nix:21]: github is a mirror

```suggestion
    homepage = "https://gitlab.com/greut/eclint";
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -32280,6 +32280,8 @@ with pkgs;
 
   eclair = callPackage ../applications/blockchains/eclair { };
 
+  eclint = callPackage ../development/tools/eclint { };
```
Comment (zowoq) [inline pkgs/top-level/all-packages.nix:32283]: all-packages is a bit of a mess be this shouldn't be added in the blockchain section, could go above `editorconfig-checker` instead.

**Comment by lucperkins** [comment]
@zowoq Good suggestions. Changes have been pushed.

**Pull request merged**